website/docs/philosophy.md

```diff
@@ -2,20 +2,146 @@
 title: Philosophy about component testing
 ---
 
-## How to test component ?
+## Philosophy of testing
 
-Before answer to this question, we need to answer to the question: "what is a component ?".
+Before introduce the way you can test your front-end application, we need to share the same definitions.
 
-A component is a Custom HTML Component that produce a view (aka. a piece of others HTML Component).
-To produce the view Component take properties and his internal state.
+## Types of tests
 
-To sum up, a component is like a JavaScript function that take two arguments, the state and the props.
+Component-test-util assume theses approximative definitions that fit well for component testing:
 
-It's ineffective to try change the state of the component manually in test.
-
-## What is the kind of test I can do with component-test-utils ?
+- _Unit Test_: Test that stay in memory. (No possible to access to HTTP, File System, ...)
+- _Integration test_: Test that can access to other things than memory (Network).
+- _End to End test_: It's basically an integration test with a special rule. No Mock or Stub allowed.
 
 You can do both unit and integration test with component test-utils.
-The only test type you cannot do is end-to-end testing because component-test-util rely on some mocking solution.
+The only test type you cannot do is end-to-end testing because component-test-util is basically a mock of your favorite framework.
```

Comment:
```suggestion
The only test type you cannot do is end-to-end testing because component-test-utils is basically a mock of your favorite framework.
```